### PR TITLE
Add simple type field auto mapping.

### PR DESCRIPTION
**Pull request by LambdAurora**
The names are read from a JSON5 file.

Here's a sample of the JSON5 file I used for testing, which we might want to see integrated in QM later on.

```json5
{
  // Codec
  "com/mojang/serialization/Codec": "codec",
  // DFU
  "com/mojang/datafixers/schemas/Schema": "schema",
  // Brigadier
  "com/mojang/brigadier/StringReader": "reader",

  // Registries
  "net/minecraft/unmapped/C_tqxyjqsk": "registry",
  "net/minecraft/unmapped/C_xhhleach": "registryKey",

  // Client
  "net/minecraft/unmapped/C_ayfeobid": "client", // MinecraftClient
  "net/minecraft/unmapped/C_cnszsxvd": "matrices", // MatrixStack
  "net/minecraft/unmapped/C_igrgeffe": "vertexConsumers", // VertexConsumerProvider
  "net/minecraft/unmapped/C_mavozmpp": "textRenderer",

  // Pos
  "net/minecraft/unmapped/C_hynzadkk": "pos", // BlockPos
  "net/minecraft/unmapped/C_hynzadkk$C_egqitdjk": {
    local_name: "pos",
    "fallback": [
      "mutablePos"
    ]
  },
  "net/minecraft/unmapped/C_ovcqqyqp": "globalPos", // GlobalPos

  // Common
  "net/minecraft/unmapped/C_rlomrsco": "random", // RandomGenerator
  "net/minecraft/unmapped/C_xpuuihxf": "direction",
  "net/minecraft/unmapped/C_eslcbfsq": "profiler",

  // States
  "net/minecraft/unmapped/C_txtbiemp": { // BlockState
    local_name: "state",
    exclusive: true,
    fallback: [
      "blockState"
    ]
  },
  "net/minecraft/unmapped/C_xqketiuf": { // FluidState
    local_name: "state",
    exclusive: true,
    fallback: [
      "fluidState"
    ]
  },

  // Block
  "net/minecraft/unmapped/C_mmxmpdoq": "block",

  // Item
  "net/minecraft/unmapped/C_sddaxwyk": "stack", // ItemStack

  // Entity
  "net/minecraft/unmapped/C_rjqjaxef": "brain",
  "net/minecraft/unmapped/C_uzzvxofv": {
    local_name: "reason",
    fallback: [
      "spawnReason"
    ]
  },

  // Resources
  "net/minecraft/unmapped/C_tmnrpasf": "resourceManager",
  "net/minecraft/unmapped/C_carvbxzj": { // AutoCloseableResourceManager
    local_name: "resourceManager",
    fallback: [
      "autoCloseableResourceManager"
    ]
  },
  "net/minecraft/unmapped/C_migzkpst": "resources", // ServerReloadableResources

  // World
  "net/minecraft/unmapped/C_cdctfzbn": "world",
  "net/minecraft/unmapped/C_agsukcmb": { // TestableWorld
    local_name: "world",
    fallback: [
      "testableWorld"
    ]
  },
  "net/minecraft/unmapped/C_bdwnwhiu": { // ServerWorld
    local_name: "world",
    exclusive: true,
    fallback: [
      "serverWorld"
    ]
  },
  "net/minecraft/unmapped/C_ghdnlrrw": { // ClientWorld
    local_name: "world",
    exclusive: true,
    fallback: [
      "clientWorld"
    ]
  }
}
```

### CODE REVIEW

**Review by ByMartrixx — COMMENTED** [review]
Looks good, aside from the comments. There's one problem we have to figure out though. When running the command to insert the mappings through a gradle daemon relative paths in the enigma profile break, and gradle is used as working dir instead of the project

File: src/main/java/org/quiltmc/enigmaplugin/index/simple_type_single/SimpleTypeSingleIndex.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+package org.quiltmc.enigmaplugin.index.simple_type_single;
+
+import cuchaz.enigma.classprovider.ClassProvider;
+import cuchaz.enigma.translation.representation.MethodDescriptor;
+import cuchaz.enigma.translation.representation.TypeDescriptor;
+import cuchaz.enigma.translation.representation.entry.ClassEntry;
+import cuchaz.enigma.translation.representation.entry.FieldEntry;
+import cuchaz.enigma.translation.representation.entry.LocalVariableEntry;
+import cuchaz.enigma.translation.representation.entry.MethodEntry;
+import org.jetbrains.annotations.Nullable;
```
[the remark this answers is not in the document]
Comment (LambdAurora) [reply src/main/java/org/quiltmc/enigmaplugin/index/simple_type_single/SimpleTypeSingleIndex.java:26]: Is it necessary to make it explicit?
json5 pulled it in.

File: src/main/java/org/quiltmc/enigmaplugin/index/simple_type_single/SimpleTypeSingleIndex.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+    public @Nullable String getField(FieldEntry fieldEntry) {
+        return this.fields.get(fieldEntry);
+    }
+
+    public @Nullable String getParam(ParameterEntry paramEntry) {
+        return this.parameters.get(paramEntry);
+    }
+
+    public List<ParameterEntry> getParamsOf(MethodEntry methodEntry) {
```
[the remark this answers is not in the document]
Comment (LambdAurora) [reply src/main/java/org/quiltmc/enigmaplugin/index/simple_type_single/SimpleTypeSingleIndex.java:75]: This method was used for debugging as the evaluate expression in debuggers sucks and cannot use lambdas :(

File: src/main/java/org/quiltmc/enigmaplugin/index/simple_type_single/SimpleTypeSingleIndex.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ByMartrixx) [reply src/main/java/org/quiltmc/enigmaplugin/index/simple_type_single/SimpleTypeSingleIndex.java:26]: The one in json5 seems quite outdated

File: src/main/java/org/quiltmc/enigmaplugin/index/simple_type_single/SimpleTypeSingleIndex.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ByMartrixx) [reply src/main/java/org/quiltmc/enigmaplugin/index/simple_type_single/SimpleTypeSingleIndex.java:75]: Ah, so this could be annotated with `@TestOnly`?

**Review by ByMartrixx — APPROVED** [review]
Looks great now, but I still have to figure out a way to fix the issue I previously mentioned

**Pull request merged**